charts/cluster-addons/templates/cni/calico.yaml

```diff
@@ -49,6 +49,73 @@ spec:
     - secret:
         name: {{ include "cluster-addons.componentName" (list . "cni-calico") }}-config
         key: overrides
+{{- if .Values.cni.calico.globalNetworkPolicy }}
+---
+apiVersion: addons.stackhpc.com/v1alpha1
+kind: Manifests
+metadata:
+  name: {{ include "cluster-addons.componentName" (list . "cni-calico-globalnetpolicy") }}
+  labels: {{ include "cluster-addons.componentLabels" (list . "cni-calico-globalnetpolicy") | nindent 4 }}
+  annotations:
+    # Tell Argo to ignore the non-controller owner references for this object
+    argocd.argoproj.io/sync-options: "ControllerReferencesOnly=true"
+spec:
+  clusterName: {{ include "cluster-addons.clusterName" . }}
+  bootstrap: true
+  targetNamespace: {{ .Values.cni.calico.release.namespace }}
+  releaseName: cni-calico-metadata-global-netpolicy
+  manifestSources:
+    - template: |
+        ---
+        apiVersion: projectcalico.org/v3
+        kind: GlobalNetworkPolicy
+        metadata:
+          name: {{ include "cluster-addons.componentName" (list . "deny-egress") }}
+        spec:
+          order: {{ .Values.cni.calico.globalNetworkPolicy.denyPriority }}
+          namespaceSelector: kubernetes.io/metadata.name != 'openstack-system'
```

Comment:
Maybe we should make this a Helm value as well, as it might depend on what the user sets for `denyEgressCidrs`?